h/services/bulk_api/group.py

```diff
@@ -0,0 +1,39 @@
+from dataclasses import dataclass
+from typing import List
+
+import sqlalchemy as sa
+from sqlalchemy.orm import Session
+
+from h.models import Annotation, Group
+from h.services.bulk_api.core import date_match
+
+
+@dataclass
+class BulkGroup:
+    authority_provided_id: str
+
+
+class BulkGroupService:
+    """A service for retrieving groups in bulk."""
+
+    def __init__(self, db: Session):
+        self._db = db
+
+    def group_search(
```

Comment:
Similar to the existing annotation_search, this takes some mandatory parameters.

This currently serves one purpose only, generating the usage report. We might make this endpoint more useful in the future making adding more parameters, making other optional...

We'll avoid any extra complexity until we get there.

Reply:
Sure. It seems like a perfectly reasonable endpoint to have anyway: h has groups, so why shouldn't it have a group search endpoint? I'm not even sure this necessarily needs to be part of the bulk API, it seems like a normal and generic enough endpoint to just be part of the normal REST API, or even an addition to the existing [group list API](https://h.readthedocs.io/en/latest/api-reference/v2/#tag/groups/paths/~1groups/get).

Bulk API is fine for now though and I suspect it's going to be easier because it uses the LMS app's authclient authentication to decide which groups it can see.

Either way for now it's a private API only used by the LMS app, so we're not committing ourselves to anything.